src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -44,10 +44,50 @@
 import scala.compat.java8.OptionConverters;
 import scala.concurrent.ExecutionContext;
 
+/**
+ * Provides a simplified interface to Mesos through USI.
+ *
+ * <p>Each connection should be a singleton. New instance are create via {@link
+ * MesosApi#getInstance(String)}.
+ */
 public class MesosApi {
 
   private static final Logger logger = LoggerFactory.getLogger(MesosApi.class);
 
+  static HashMap<String, MesosApi> sessions = new HashMap<>();
```

Comment:
Is this a HashMap because one could create multiple MesosCloud's from one Jenkins master?

Reply:
Yes. The new plugin supports multiple `MesosCloud` configuration. It actually was a feature request by a customer at some point.